server.js

```diff
@@ -24,11 +21,162 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+/////////////////SCHEMA AND MODEL//////////////////////////
+const bookSchema = new Schema({
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number, //original input was 4.38, don't know if I have to define this as a number with decimals, do you have to? hmm....
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+
+}) //this is the schema of the book 
+
+const Book = mongoose.model("Book", bookSchema);
+//Book is a model that represents a collection of documents. It is a way to interact with the "books" collection. It can be used for CRUD. 
+
+/////////////// RESET/SEED DATABASE USING ENVIRONMENTAL VARIABLE //////////////
+if (process.env.RESET_DB) {
+  // if the environmental variable RESET_DB is used it resets the database to contain the data that is in the booksData (books.json).
+  console.log('Resetting database!')
+  // we have a consle log that tells us what's happening 
+	const seedDatabase = async () => {
+    // the seedDatabase is started and also has async, which tells it to wait for some processes to be finished.
+    await Book.deleteMany({})
+    //awaut tells the function seedDatabase to wait for the deletion of all books in the database (using the mongoose function deleteMany). The empty object passed to deleteMany is a filter object, it will match ALL documents in the collection. 
+  }
+
+  seedDatabase()
+  //The function is invoked.
+}
+
+
+//////////////ROUTES//////////////////////
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const bookAPIGuide = {
+    Routes: [
+      { 'Hello': 'Welcome to Annikas book-API! See instrucktions below',
+        '/books': 'Get all books.',
+        '/books?authors=[authorname]': 'Get all books with that author. Be aware that authors is plural.',
+        '/books?title=[title]': 'Get all books with that title, you can write a part of the title.',
+        '/books/?title=[title]&authors=[authorname]': 'Get books with a specific author and title. You can write part of the authors name',
+        '/books/:bookID': 'Get specific book by ID',
+        '/isbn/:isbn': 'Get a book`s isbn',
+      },
+    ],
+  };
+  res.send({responseMessage: bookAPIGuide});
 });
 
+
+app.get("/books", async (req, res) => {
+  const {title, authors} = req.query;
+  // we define that we can search for title and authors, seperately or combined. 
+  const response = {
+    success: true, 
+    body: {} // body is an empty object that will hold the response data.
+  }
+  // response is an object that we use to hold the response data that we want to send back to the client in the endpoint /books. 
+
+  const authorsRegex = new RegExp(authors, "i")
```

Comment:
I did not follow along with the coding session when they did this, and will be following up in my own code. Seems to make it easier for finding and replacing, as well as splicing!